pallets/collator-selection/src/lib.rs

```diff
@@ -178,11 +190,14 @@ pub mod pallet {
 	pub type Candidates<T: Config> =
 		StorageValue<_, Vec<CandidateInfo<T::AccountId, BalanceOf<T>>>, ValueQuery>;
 
-	/// Last block authored by collator.
+	pub(super) type BlockCount = u32;
+	#[pallet::type_value]
+	pub(super) fn StartingBlockCount() -> BlockCount {
+		0u32.into()
+	}
 	#[pallet::storage]
-	#[pallet::getter(fn last_authored_block)]
-	pub type LastAuthoredBlock<T: Config> =
-		StorageMap<_, Twox64Concat, T::AccountId, T::BlockNumber, ValueQuery>;
+	pub(super) type BlocksPerCollatorThisSession<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::AccountId, BlockCount, ValueQuery, StartingBlockCount>; // RAD: Note: AccountId is user-selectable, but protected by ED, is that enough?
```

Comment:
AccountId is user-selectable, but protected by ED, is that enough? to use the non cryptographic hashing algo?

Reply:
sticking with Blake2 as the storage item will be small in size ( < 1000 ) so performance impact should be negligible